src/main/webapp/index.html

```diff
@@ -35,39 +35,42 @@ <h1 class="index-registration-block-child">TravIS: Travel Information System</h1
         <form action="/calculate-trip" method="POST" id="startTripForm">
           <div class="form-row">
             <div class="form-group col-md-3">
-              <input type="text" placeholder="Destination" class="form-control" 
-                name="inputDestination" id="inputDestination">
+              <label for="inputDestination">Destination</label>
+              <input type="text" placeholder="Where to?" class="form-control" 
+                name="inputDestination" id="inputDestination" 
+                onchange="checkValidInput('inputDestination'); checkNextButton();
+                  checkSubmitButton()">
             </div>
             <div class="form-group col-md-3">
-              <input type="text" placeholder="Start Date" class="form-control" 
-                name="inputStartDate" id="inputStartDate">
+              <label for="inputStartDate">Start Date</label>
+              <input type="date" class="form-control" name="inputStartDate" 
+                id="inputStartDate" oninput="setEndDateValue()"
+                onchange="checkValidInput('inputStartDate'); checkNextButton(); 
+                  checkSubmitButton()">
             </div>
             <div class="form-group col-md-3">
-              <input type="text" placeholder="End Date" class="form-control is-valid" 
-                name="inputEndDate" id="inputEndDate" readonly>
+              <label for="inputEndDate">End Date</label>
+              <input type="date" class="form-control" name="inputEndDate" 
+                id="inputEndDate" readonly>
             </div>
             <div class="form-group">
-              <button type="button" class="btn btn-secondary">Next</button>
+              <input type="button" class="btn btn-success" value="Next" 
+                id="toggle-stage-button" onclick="toggleStartTripInputStage()">
             </div>
           </div>
-          <div class="form-row">
+          <div class="form-row" id="add-pois-container">
```

Comment:
Optional: my personal preference would be for the POI's I've already entered to stack up vertically instead of line-wrapping as they currently do. Eg vertically stacking these elements: https://screenshot.googleplex.com/whUAXWbqF58

Reply:
I like that too -- done!